gsy_framework/schemas.py

```diff
@@ -202,8 +208,32 @@ class ScenarioSchemas:
                     "max_available_power_kW":  {"anyOf": [{"type": "number"}, {"type": "null"}]}
                 }
             },
+            "heat_pump": {
+                "type": "object",
+                "properties": {
+                    "name": {"type": "string"},
+                    "type": {"enum": ["HeatPump"]},
+                    "uuid": {"type": "string"},
+                    "libraryUUID": {"anyOf": [{"type": "string"}, {"type": "null"}]},
+                    "maximum_power_rating_kW": {"type": "number"},
+                    "min_temp_C": {"type": "number"},
+                    "max_temp_C": {"type": "number"},
+                    "initial_temp_C": {"type": "number"},
+                    "external_temp_C": {"anyOf": [{"type": "number"}, {"type": "null"}]},
+                    "external_temp_profile_uuid": {
+                        "anyOf": [{"type": "string"}, {"type": "null"}]},
+                    "tank_volume_l": {"type": "number"},
+                    "consumption_kW": {"anyOf": [{"type": "number"}, {"type": "null"}]},
+                    "consumption_profile_uuid": {"anyOf": [{"type": "string"}, {"type": "null"}]},
+                    "source_type": {
+                        "enum": [HeatPumpSourceType.AIR.value, HeatPumpSourceType.GROUND.value]},
```

Comment:
interesting. Did not know that there is an enum field in JSON schema

Reply:
Admitedly, JsonSchema is much more powerful than AVRO. It has some mind-blowing features.

Reply:
Why did we decided for AVRO then? ;-)

Reply:
Mostly because of it's binary encoding and better compatibility with Kafka. However we can still debate using of AVRO over JsonSchema.